source/geod24/concurrency.d

```diff
@@ -897,6 +898,78 @@ class FiberScheduler
         return ThreadInfo.thisInfo;
     }
 
+    final static class FiberBinarySemaphore
+    {
+
+        /***********************************************************************
+
+            Associate `FiberBinarySemaphore` with the running `Fiber`
+
+            `FiberScheduler` will check to see if the `Fiber` is blocking on a
+            `FiberBinarySemaphore` to avoid rescheduling it unnecessarily
+
+        ***********************************************************************/
+
+        private void registerToInfoFiber(InfoFiber info_fiber = cast(InfoFiber) Fiber.getThis()) nothrow
+        {
+            assert(info_fiber !is null, "This Fiber does not belong to FiberScheduler");
+            assert(info_fiber.sem is null, "This Fiber already has a registered FiberBinarySemaphore");
+            info_fiber.sem = this;
+
+        }
+
+        void wait() nothrow
```

Comment:
Style nitpick: We always put space between a parenthesis and a function declaration or definition.
So this should be:
```suggestion
        void wait () nothrow
```

On the other hand, if you call it, no space:
```D
foo.wait();
```

This makes grepping / Ctrl + F for the definition super simple.

Reply:
Noted.

Reply:
Applied this style to functions I recently touched or wrote myself.